File: BugTrackerII-FE/ngfrontEnd/src/app/model/bug.ts
export class Bug {
    public id!: number;
    public projectName!: string;
    public issueType!: string;
    public description!: string;
    public troubleShooting!: string;
    public resolution!: string;
    public resolved!: boolean;
    public recordedDate!: Date;
    public updateDate!: Date; 
    
   
    static displayGetBug(id:number, name:string, type: string, desc:string, 
        tsing:string, resolved:boolean, resolution: string, rdate: Date, udate: Date ):Bug{
         var bug = new Bug();
        bug.id = id;
        bug.projectName = name;
        bug.issueType = type;
        bug.description = desc;
        bug.troubleShooting= tsing;
        bug.resolved = resolved;
        bug.resolution = resolution;
        bug.recordedDate = rdate;
        bug.updateDate = udate;
        return bug;
    }
    static postBug(name: string, type: string, desc:string, 
         tsing:string, resolved:boolean, resolution:string ):Bug{
        var post = new Bug();
        post.projectName = name;
        post.issueType = type;
        post.description = desc;
        post.troubleShooting= tsing;
        post.resolved = resolved;
        post.resolution = resolution;
        return post;
    }

    static putBug(id:number, name:string, type: string, desc:string, 
        tsing:string, resolved:boolean, resolution: string,):Bug{
            var bug = new Bug();
        bug.id = id;
        bug.projectName = name;
        bug.issueType = type;
        bug.description = desc;
        bug.troubleShooting= tsing;
        bug.resolved = resolved;
        bug.resolution = resolution;
        return bug;
        }
    
   
}

File: BugTrackerII-BE/bugTracker/src/main/java/com/electratech/bugTracker/entity/Bug.java
package com.electratech.bugTracker.entity;

import java.util.Date;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Bug {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String projectName;
	private String issueType;
	private String description;
	private String troubleShooting;
	private boolean isResolved;
	private String resolution;
	@CreationTimestamp
	private Date recordedDate;
	@UpdateTimestamp
	private Date updateDate;
	
	
	public Bug(String projectName, String issueType, String description, String troubleShooting, boolean isResolved,
			String resolution) {
		super();
		this.projectName = projectName;
		this.issueType = issueType;
		this.description = description;
		this.troubleShooting = troubleShooting;
		this.isResolved = isResolved;
		this.resolution = resolution;
		
	}
	
	@Override
	public String toString() {
		return "Bug [projectName=" + projectName + ", issueType=" + issueType + ", Description=" + description
				+ ", troubleShooting=" + troubleShooting + ", isResolved=" + isResolved + ", resolution=" + resolution
				+ ", recordedDate=" + recordedDate + ", updateDate=" + updateDate + "]";
	}
	
	
}
